src/State/Hub.php

```diff
@@ -169,6 +172,31 @@ public function captureLastError(?EventHint $hint = null): ?EventId
         return null;
     }
 
+    public function captureCheckIn(string $slug, MonitorConfig $upsertMonitorConfig, CheckInStatus $status, ?CheckIn $previous = null): ?CheckIn
+    {
+        $client = $this->getClient();
+
+        if (!$client instanceof ClientInterface) {
+            return null;
+        }
+
+        $options = $client->getOptions();
+        $event = Event::createCheckIn();
+        $checkIn = new CheckIn(
+            $slug,
+            $status,
+            $previous ? $previous->getId() : null,
```

Comment:
As per above, I think it's always better to be explicit with checks: in this case, we really want to make it clear to the reader that `$previous` must not be `null`, so I would change the code accordingly